samples/DispatcherSample.Web/Startup.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DispatcherSample.Web
+{
+    public class Startup
+    {
+        private static readonly byte[] _helloWorldPayload = Encoding.UTF8.GetBytes("Hello, World!");
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddRouting();
+            services.AddDispatcher(options =>
+            {
+                options.DataSources.Add(new DefaultEndpointDataSource(new[]
+                {
+                    new MatcherEndpoint((next) => (httpContext) =>
+                    {
+                        var response = httpContext.Response;
+                        var payloadLength = _helloWorldPayload.Length;
+                        response.StatusCode = 200;
+                        response.ContentType = "text/plain";
+                        response.ContentLength = payloadLength;
+                        return response.Body.WriteAsync(_helloWorldPayload, 0, payloadLength);
+                    },
+                    "/plaintext", new { }, 0, EndpointMetadataCollection.Empty, "Plaintext"),
+                }));
```

Comment:
This is intentionally kinda ugly. 

The plan for 2.2 is that there is no mechanism for **application developers** to add endpoints (other than MVC). We can add endpoints using these kinds of a primitives, but it's not meant to be easier or more fun than it is right here.

This is because we're not trying to make an alternative to MVC or middleware

Reply:
I'm open to more discussion about this but what I'm trying to say is that I think it's specifically a non goal. We don't have the other tools that users will need to build real applications in an expressjs style.